### PR TITLE
Add Assignment List UI

### LINKED ISSUE #76: As a conscientious Avenger, I can view the grades of my students for their assignments
As a conscientious Avenger, I can view the grades of my students for their assignments, allowing me to better analyze how they are performing with their work.

### LINKED ISSUE #94: As a conscientious Avenger, I can view the comments of my students for their assignments
As a conscientious Avenger, I can view the comments of my students for their assignments, allowing me to better take note of mistakes they make.

### LINKED ISSUE #95: As a busy Avenger, I can view all assignment names
As a busy Avenger, I can view all assignment names, helping me to save time finding the assignment names.

### PR DESCRIPTION
**Pull request by Originalidk**
Add a list of assignment names to the UI.

### CODE REVIEW

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -92,6 +95,12 @@ public ObservableList<Session> getFilteredSessionList() {
         return model.getFilteredSessionList();
     }
 
+    @Override
+    public ObservableList<AssignmentName> getAssignmentNameList() {
+        AssignmentInitialise.init();
+        return AssignmentNameInitialise.getAllNames();
```
Comment (ldinghan) [inline src/main/java/seedu/address/logic/LogicManager.java:113]: Could add a test case covering this line

File: src/main/java/seedu/address/model/person/assignment/initialise/AssignmentNameInitialise.java [hunk 2]
```diff
@@ -49,6 +50,10 @@ public AssignmentName getName(int index) {
         return assignmentNames.get(index);
     }
 
+    public static ObservableList<AssignmentName> getAllNames() {
+        return assignmentNames;
```
Comment (ldinghan) [inline src/main/java/seedu/address/model/person/assignment/initialise/AssignmentNameInitialise.java:54]: test case!

**Review by ldinghan — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/LogicManager.java (hunk 1 above)
Comment (raydenlim) [reply src/main/java/seedu/address/logic/LogicManager.java:113]: I agree. Codecov will be happier like this

File: src/main/java/seedu/address/model/person/assignment/initialise/AssignmentNameInitialise.java (hunk 2 above)
Comment (raydenlim) [reply src/main/java/seedu/address/model/person/assignment/initialise/AssignmentNameInitialise.java:54]: i second

File: src/main/java/seedu/address/model/person/assignment/initialise/AssignmentNameInitialise.java [hunk 3]
```diff
@@ -49,6 +50,10 @@ public AssignmentName getName(int index) {
...
+    }
+
     public boolean contains(AssignmentName assignmentName) {
```
Comment (raydenlim) [inline src/main/java/seedu/address/model/person/assignment/initialise/AssignmentNameInitialise.java:57]: wait, this method seems redundant. Not sure if i am high... but...
contain contains.

File: src/main/java/seedu/address/ui/AssignmentNameListPanel.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            super.updateItem(name, empty);
+
+            if (empty || name == null) {
+                setGraphic(null);
+                setText(null);
+            } else {
+                setGraphic(new AssignmentNameCard(name).getRoot());
+            }
+        }
+    }
```
Comment (raydenlim) [inline src/main/java/seedu/address/ui/AssignmentNameListPanel.java:49]: Please write testcase for the "}" 
Codecov not happy :(

**Review by raydenlim — COMMENTED** [review]
LGTM. Clean and Green

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 5]
```diff
@@ -19,21 +19,30 @@ public class CommandResult {
     /** The application should exit. */
     private final boolean exit;
 
+    /** The application should show a person's assignments. */
+    private final boolean personAssignments;
+
+    /** The application should show assignment names. */
+    private final boolean assignmentNames;
+
```
Comment (wesho1107) [inline src/main/java/seedu/address/logic/commands/CommandResult.java:27]: Perhaps could follow the standard naming conventions for boolean variables.

File: src/test/java/seedu/address/logic/commands/CommandResultTest.java [hunk 6]
```diff
@@ -14,7 +14,7 @@ public void equals() {
 
         // same values -> returns true
         assertTrue(commandResult.equals(new CommandResult("feedback")));
-        assertTrue(commandResult.equals(new CommandResult("feedback", false, false)));
+        assertTrue(commandResult.equals(new CommandResult("feedback", false, false, false, false)));
```
Comment (wesho1107) [inline src/test/java/seedu/address/logic/commands/CommandResultTest.java:17]: Perhaps you could use a static constant for "feedback" and "different".

**Review by wesho1107 — COMMENTED** [review]
Other than a few suggestions, LGTM!

**Pull request merged**